packages/core-ethereum/src/channel.ts

```diff
@@ -61,13 +61,11 @@ class Channel {
 
     const ticket = unacknowledgedTicket.ticket
 
-    if (ticket.isWinningTicket(await this.commitment.getCurrentCommitment(), response, ticket.winProb)) {
-      const ack = new AcknowledgedTicket(
-        ticket,
-        response,
-        await this.commitment.getCurrentCommitment(),
-        unacknowledgedTicket.signer
-      )
+    // @TODO use some caching to optimize execution time
+    const opening = await this.commitment.findPreImage(await this.commitment.getCurrentCommitment())
+
+    if (ticket.isWinningTicket(opening, response, ticket.winProb)) {
+      const ack = new AcknowledgedTicket(ticket, response, opening, unacknowledgedTicket.signer)
```

Comment:
Why did the types not prevent this bug?

Reply:
It's a logical error - at least as far as I see it. Commitment and Opening are currently the same type, so only giving Opening a different type will prevent from this error